starlite/contrib/repository/abc.py

```diff
@@ -0,0 +1,167 @@
+"""Data persistence interface."""
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
+
+from .exceptions import NotFoundError
+
+if TYPE_CHECKING:
+    from .types import FilterTypes
+
+__all__ = ["AbstractRepository"]
+
+T = TypeVar("T")
+RepoT = TypeVar("RepoT", bound="AbstractRepository")
+
+
+class AbstractRepository(Generic[T], metaclass=ABCMeta):
+    """Interface for persistent data interaction."""
+
+    model_type: type[T]
+    """Type of object represented by the repository."""
+    id_attribute = "id"
+    """Name of the primary identifying attribute on `model_type`."""
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Repository constructors accept arbitrary kwargs."""
+        super().__init__(**kwargs)
+
+    @abstractmethod
+    async def add(self, data: T) -> T:
+        """Add `data` to the collection.
+
+        Args:
+            data: Instance to be added to the collection.
+
+        Returns:
+            The added instance.
+        """
+
+    @abstractmethod
+    async def delete(self, id_: Any) -> T:
```

Comment:
Instead of `id_` why not `identifier` or `item_id`?

I think forward underscore looks bad and documents bad